File: mktrs-spring-boot-native-java-poc/src/main/java/com/themusketeers/sbnative/domain/response/UserDataResponse.java
/*----------------------------------------------------------------------------*/
/* Source File:   USERDATARESPONSE.JAVA                                       */
/* Copyright (c), 2023 The Musketeers                                         */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 May.30/2023  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.themusketeers.sbnative.domain.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.themusketeers.sbnative.domain.User;

/**
 * Keeps the users for the User List response.
 *
 * @param user Indicates one user information record retrieved.
 * @author COQ - Carlos Adolfo Ortiz Q.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({"user"})
public record UserDataResponse(User user) {
}

File: mktrs-spring-boot-native-kotlin-poc/src/main/kotlin/com/themusketeers/sbnative/domain/response/UserDataResponse.kt
/*----------------------------------------------------------------------------*/
/* Source File:   USERDATARESPONSE.JAVA                                       */
/* Copyright (c), 2023 The Musketeers                                         */
/*----------------------------------------------------------------------------*/
/*-----------------------------------------------------------------------------
 History
 May.30/2023  COQ  File created.
 -----------------------------------------------------------------------------*/
package com.themusketeers.sbnative.domain.response

import com.themusketeers.sbnative.domain.User

/**
 * Keeps the users for the User List response.
 *
 * @param count Indicates how many users are registered in the system.
 * @param users Indicates the registered User List.
 * @author COQ - Carlos Adolfo Ortiz Q.
 */
data class UserDataResponse(val count: Long, val users: List<User>)
